superset/assets/javascripts/modules/superset.js

```diff
@@ -222,14 +222,19 @@ const px = function () {
         timer = setInterval(stopwatch, 10);
         $('#timer').removeClass('label-danger label-success');
         $('#timer').addClass('label-warning');
-        this.viz.render();
+        $.getJSON(this.jsonEndpoint(), queryResponse => {
+          try {
+            vizMap[data.form_data.viz_type](this, queryResponse);
+            this.done(queryResponse);
```

Comment:
curious about the case when `this.done` would fail, which makes the `try/catch` neccessary... could you elaborate?